### DISCUSSION TITLE
Support live-reloading of tailwind config

### DISCUSSION THREAD

**Discussion by itsMapleLeaf** [ben-rogerson/twin.macro#37] [Ideas]
Right now, I need to restart the webpack dev server whenever I make a change to the tailwind config. It'd be nice if I could make a change in the config, and those changes would be reflected without the need for a server restart. Making the issue here, though I'm not sure if this issue is best fixed someplace else

**Comment by ben-rogerson** [comment]
Hey daro, yeah that's annoying. I'm working on hot reloading the config right now 👍 

<img src="https://user-images.githubusercontent.com/21288568/79160014-3d3dda00-7e18-11ea-977e-d72469f4803f.png" width="300">

From https://github.com/ben-rogerson/twin.macro/issues/11:
> tailwind.macro - the package twin was forked from, has the ability to pick up live tailwind.config.js changes without a build script restart. I removed that code and now I'm working on re-incorporating it.
> Shouldn't be too long if all goes smoothly 🤞

**Comment by ben-rogerson** [comment]
Heads up, I'm putting a freeze on this feature while I tackle [adding the features](https://github.com/ben-rogerson/twin.macro/issues/45) from the new Tailwind release.

**Comment by j1mr10rd4n** [comment]
I might be able to help out with this. After doing some digging, would you agree the following is along the right track?

Looks like some progress is being made on watching ancillary source files for changes and triggering recompilation (see [Babel PR #11741 - Allow plugins to indicate dependencies on random files](https://github.com/babel/babel/pull/11741).

The PR hasn't been merged yet and I think it would need to go together with a change to [babel-loader](https://github.com/babel/babel-loader) so that it could register watched files with webpack and the webpack devserver.

**Comment by ben-rogerson** [comment]
Hey thanks for picking up the thread on this.
I think a good solution would be one that would benefit everyone. Would this target people using webpack bundlers only?

**Comment by vedantroy** [comment]
[That PR](https://github.com/babel/babel/pull/11741) adds an API to Babel's plugin system, which could be used in twin.macro. Out of the box, it benefits people who use Babel CLI's --watch option.

Consumers of Babel (Rollup, Webpack, etc.) would have to write integrations with that PR in order to reap the benefits.

**Comment by MauriceArikoglu** [comment]
@ben-rogerson @vedantroy Any updates on this one? I know @vedantroy did a great job on the PR, but it seems the babel team didn't merge yet...
